### ISSUE TITLE
Incorrect error reported when validating XHTML image map

### ISSUE THREAD

**Issue by raducoravu** [IDPF/epubcheck#696]
If in the XHTML inside the EPUB I have an image map specified something like:

```
  <img usemap="#d179e39" src="../images/Gear_pump_exploded.png" alt="Gear Pump" />
     <map name="d179e39" id="d179e39">
   ..........
```

EPUB validation reports an error like:

```
   Description: Error while parsing file 'value of attribute "usemap" is invalid; must be an XML name without colons'.
```

but actually according to the specs the @usemap attribute must use a hash symbol to refer to the map name:

https://www.w3.org/TR/html5/embedded-content-0.html#attr-hyperlink-usemap

**Comment by tofi86** [comment]
This is valid. There's a difference between XHTML 1.1 (EPUB 2) and HTML 5 (EPUB 3) syntax for image maps:

**XHTML 1.1** uses `ID`/`IDREF` datatypes for `@id`/`@usemap` and therefore doesn't need the `#`:
```html
<img src="images/map.jpg" usemap="imagemap" width="479" height="312" alt="" />
<map id="imagemap">
	<area shape="rect" coords="55,101,129,284" href="target.xhtml#t1" alt="Touch" title="Touch" />
	<area shape="rect" coords="153,23,270,279" href="target.xhtml#t2" alt="Touch" title="HDPlus" />
	<area shape="rect" coords="273,79,395,273" href="target.xhtml#t3" alt="Touch" title="Touch" />
</map>
```

**HTML 5** uses a special datatype for `@name`/`@usemap` which requires the `#`:
```html
<img src="images/map.jpg" usemap="#imagemap" width="479" height="312" alt="" />
<map name="imagemap">
	<area shape="rect" coords="55,101,129,284" href="target.xhtml#t1" alt="Touch" title="Touch" />
	<area shape="rect" coords="153,23,270,279" href="target.xhtml#t2" alt="Touch" title="HDPlus" />
	<area shape="rect" coords="273,79,395,273" href="target.xhtml#t3" alt="Touch" title="Touch" />
</map>
```

So I think you might have checked HTML 5 syntax in an XHTML 1.1 document (watch the DOCTYPE!!).

**Comment by tofi86** [comment]
To make this clear for the future, I added 4 single tests with 31bd74e in the OPS testsuite.

**Comment by tofi86** [comment]
> So I think you might have checked HTML 5 syntax in an XHTML 1.1 document (watch the DOCTYPE!!).

P.S.: this could have also been a missing DOCTYPE declaration...

**Comment by mattgarrish** [comment]
But they fixed that ID/IDREF problem in the last release of XHTML 1.1 (see the third item in the [status section](https://www.w3.org/TR/xhtml-modularization/Overview.html#status)).

It's supposed to be a URIREF now, so I think in this case that EPUB 2 validation does need to be fixed.

**Comment by mattgarrish** [comment]
Interestingly, OPS 2.01 uses the old syntax in [s. 2.6.3.1.2.2](http://www.idpf.org/epub/20/spec/OPS_2.0.1_draft.htm#Section2.6.3.1.2.2).

I'm guessing no one spotted the change before the release. It seems problematic to revise now given that this change isn't backwards compatible, but starting with a hash is the correct syntax and OPS references undated versions of 1.1 and the modularization spec.

I'd be surprised if image maps are a very common feature of epub 2s, however.

**Comment by tofi86** [comment]
Hey Matt, thanks for having a second look at this — this really seems to be a big mess...

> But they fixed that ID/IDREF problem in the last release of XHTML 1.1 (see the third item in the status section).

Okay, got it. So for XHTML _1.1_ ([DTD link](https://www.w3.org/TR/xhtml11/DTD/xhtml11-flat.dtd)) the correct use would be `usemap="#map"` with corresponding `<map id="map">`.

So here come the problems:

> It's supposed to be a URIREF now, so I think in this case that EPUB 2 validation does need to be fixed.

Yes that's true. However, when looking at the current EpubCheck XHTML DTDs ([Strict](https://github.com/IDPF/epubcheck/blob/master/src/main/resources/com/adobe/epubcheck/schema/20/dtd/xhtml1-strict.dtd)) and [Transitional](https://github.com/IDPF/epubcheck/blob/master/src/main/resources/com/adobe/epubcheck/schema/20/dtd/xhtml1-transitional.dtd)) that the `usemap` attribute _IS_ already a URI(REF) datatype, whereas the document header states that the document itself is the XHTML 1.0 (! ZERO) DTD. That's clearly a mismatch.

Then there's the RNG XHTML schema which defines the `usemap` datatype as `IDREF` in [`csismap.rng`](https://github.com/IDPF/epubcheck/blob/master/src/main/resources/com/adobe/epubcheck/schema/20/rng/xhtml/csismap.rng).

To sum this up: We obviously have an outdated (and manually edited) XHTML DTD version in EpubCheck which is not used for validation anyway (??) because the RNG XHTML schema (which is outdated as well) is used instead. I opened issue #733 to fix this.


> Interestingly, OPS 2.01 uses the old syntax in s. 2.6.3.1.2.2.

Nope, the example uses a totally _mixed_ and invalid syntax: `usemap` attribute with IDREF token (old XHTML syntax) and a `<map>` element with `name` attribute which isn't valid in XHTML at all (needs to be an `id` attribute).

> It seems problematic to revise now given that this change isn't backwards compatible, but starting with a hash is the correct syntax and OPS references undated versions of 1.1 and the modularization spec.

I would at least revise the EpubCheck schemas in #733. But I would also suggest to revise the OPS docs since the example is totally invalid as stated above...

> I'd be surprised if image maps are a very common feature of epub 2s, however.

Sure, I agree. But at least this discussion lead to the discovery that the EpubCheck schemas never got updated...

Speeking of schema updates, there's obviously more to be done: OPF needs an update (https://github.com/IDPF/epubcheck/pull/725) and probably others too...

/cc @rdeltour 

----

@raducoravu speaking of the latest XHTML 1.1 schema revision which @mattgarrish linked above: the test documents I committed validate just fine in oXygen 18.1. Which means that you're also not shipping the latest XHTML schemas in oXygen at the moment... Would you consider an update?

**Comment by tofi86** [comment]
I updated my comment above because I was wrong with the following section:

> It's supposed to be a URIREF now, so I think in this case that EPUB 2 validation does need to be fixed.

~~Yes that's true. However, when looking at the current EpubCheck XHTML DTDs ([Strict](https://github.com/IDPF/epubcheck/blob/master/src/main/resources/com/adobe/epubcheck/schema/20/dtd/xhtml1-strict.dtd)) and [Transitional](https://github.com/IDPF/epubcheck/blob/master/src/main/resources/com/adobe/epubcheck/schema/20/dtd/xhtml1-transitional.dtd)) that the `usemap` attribute _IS_ already a URI(REF) datatype, whereas the document header states that the document itself is the XHTML 1.0 (! ZERO) DTD. That's clearly a mismatch.~~

I was mistakenly referring to the XHTML 1.0 DTD where the `usemap` datatype has ever been `URI(REF)`.

**Comment by mattgarrish** [comment]
> Nope, the example uses a totally mixed and invalid syntax: usemap attribute with IDREF token (old XHTML syntax) and a <map> element with name attribute which isn't valid in XHTML at all (needs to be an id attribute).

Right, good catch. I wasn't even looking closely at the map element.

I've sort of given up reporting issues with epub 2, though, as the last one I opened just languished for five years in the tracker until I gave up and closed it off. I don't know that those docs will ever get another update, and there are no errata links in them that we could add to.

**Comment by tofi86** [comment]
> I've sort of given up reporting issues with epub 2, though, as the last one I opened just languished for five years in the tracker until I gave up and closed it off. I don't know that those docs will ever get another update, and there are no errata links in them that we could add to.

Okay, I understand that... I think that glitch is acceptable in the specs, as it's just an example...
However, I would still like to update the RNG schema in #733 as still a lot of people use and check EPUB2.

**Comment by mattgarrish** [comment]
Ya, unfortunately I can't help you with the question of where those schemas came from or what might have been done to them. Before my time.

**Comment by raducoravu** [comment]
> speaking of the latest XHTML 1.1 schema revision which @mattgarrish linked above: the test documents I committed validate just fine in oXygen 18.1. Which means that you're also not shipping the latest XHTML schemas in oXygen at the moment... Would you consider an update?

@tofi86 I'm kind of confused what I need to do in Oxygen, if you write me an email with a sample document and some details about what you do with it (open it in Oxygen and validate or use the EPUB Checker to validate the entire EPUB in which the document is located) I could try to look into this.

**Comment by tofi86** [comment]
@raducoravu will write an email, sure... :)

**Comment by tofi86** [comment]
The XHTML 1.1 RNG schema will be updated to the latest version with PR #747 in milestone 4.1
